### PR TITLE
Add   hide notes support

### PR DESCRIPTION
**Pull request by yxxhero**
https://github.com/helm/helm/pull/11600

### CODE REVIEW

**Review by zhaque44 — APPROVED** [review]
This looks good to me

File: pkg/state/helmx.go [hunk 1]
```diff
@@ -119,6 +119,22 @@ func (st *HelmState) appendCascadeFlags(flags []string, helm helmexec.Interface,
...
+	// see https://github.com/helm/helm/releases/tag/v3.16.0
+	if !helm.IsVersionAtLeast("3.16.0") {
+		return flags
+	}
+	switch {
+	// postRenderer arg comes from cmd flag.
+	case ops.HideNotes:
+		flags = append(flags, "--hide-notes")
+	case st.HelmDefaults.HideNotes:
+		flags = append(flags, "--hide-notes")
```
Comment (mumoshu) [inline pkg/state/helmx.go:133]: Let's remove this and keep it a helmfile command-flag, until we are sure there IS an use-case where you want to toggle it per release.

For now, I consider this as an operation concern that should be toggled at helmfile runtime.

File: docs/index.md [hunk 2]
```diff
@@ -236,6 +236,8 @@ helmDefaults:
   # suppressOutputLineRegex is a list of regex patterns to suppress output lines from helm diff (default []), available in helmfile v0.162.0 
   suppressOutputLineRegex:
     - "version"
+  # --hide-notes flag for helm, available in helmfile v0.169.0 with helm v3.16.0+, (default false)
+  hideNotes: false
```
Comment (mumoshu) [inline docs/index.md:240]:
```suggestion
```

File: pkg/state/state.go [hunk 3]
```diff
@@ -205,6 +205,8 @@ type HelmSpec struct {
 	DeleteWait bool `yaml:"deleteWait"`
 	// Timeout is the time in seconds to wait for helmfile delete command (default 300)
 	DeleteTimeout int `yaml:"deleteTimeout"`
+	// HideNotes, if set to true, will hide notes
+	HideNotes bool `yaml:"hideNotes"`
```
Comment (mumoshu) [inline pkg/state/state.go:209]:
```suggestion
```

**Review by mumoshu — COMMENTED** [review]
Great job!
One thing- I believe we keep it simple by treating hide-flags a command-line-flag-only option for now.

**Review by mumoshu — APPROVED** [review]
LGTM!

**Pull request merged**